src/osgPlugins/dae/daeRGeometry.cpp

```diff
@@ -470,22 +454,22 @@ void daeReader::processSinglePPrimitive(osg::Geode* geode,
     domP_Array domPArray;
     domPArray.append(group->getP());
     std::vector<std::vector<GLuint> > indexLists;
-    resolveMeshArrays(domPArray, group->getInput_array(), pDomMesh, geometry.get(), sources, indexLists);
+    resolveMeshArrays(domPArray, group->getInput_array(), pDomMesh, geometry, sources, indexLists);
     if (!indexLists.front().empty())
```

Comment:
`geometry` leaks if this is false

Use `ref_ptr` like before but `geometry.release()` instead of `geometry.get()`?

Reply:
"If this is false" - what do you mean? resolveMeshArrays is of type `void`, and above, `osg::Geometry* geometry = new osg::Geometry;` shouldn't return a nullptr. Maybe I'm not noticing something here, but please explain. One point of view is that if someone changes the code, perhaps ref_ptr can be safer?

Reply:
if the code doesn't go into the if branch, the ownership of `geometry` does not get transferred into `geometryGroup`, so it never gets deleted.

As I understand, the previous version was buggy in another way - `geometry` was deleted immediately after the function finished.

Or perhaps `geometryGroup->addChild` does not transfer ownership, in which case the previous code was correct.